src/Tools/dotnet-monitor/Auth/MonitorApiKeyPostConfigure.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    /// <summary>
+    /// Sets options on <see cref="MonitorApiKeyConfiguration"/> based on <see cref="MonitorApiKeyOptions"/>.
+    /// This class is responsible for decoding the value provided in <see cref="MonitorApiKeyOptions.PublicKey"/> and validating it.
+    /// </summary>
+    internal sealed class MonitorApiKeyPostConfigure :
+        IPostConfigureOptions<MonitorApiKeyConfiguration>
+    {
+        private readonly ILogger<MonitorApiKeyPostConfigure> _logger;
+        private readonly IOptionsMonitor<MonitorApiKeyOptions> _apiKeyOptions;
+
+        public MonitorApiKeyPostConfigure(
+            ILogger<MonitorApiKeyPostConfigure> logger,
+            IOptionsMonitor<MonitorApiKeyOptions> apiKeyOptions)
+        {
+            _logger = logger;
+            _apiKeyOptions = apiKeyOptions;
+        }
+
+        public void PostConfigure(string name, MonitorApiKeyConfiguration options)
+        {
+            MonitorApiKeyOptions sourceOptions = _apiKeyOptions.CurrentValue;
+
+            IList<ValidationResult> errors = new List<ValidationResult>();
+
+            Validator.TryValidateObject(
+                sourceOptions,
+                new ValidationContext(sourceOptions, null, null),
+                errors,
+                validateAllProperties: true);
+
+            if (string.IsNullOrEmpty(sourceOptions.Subject))
+            {
+                errors.Add(
+                    new ValidationResult(
+                        string.Format(
+                            Strings.ErrorMessage_FieldNotAllowed,
+                            nameof(MonitorApiKeyOptions.Subject),
+                            sourceOptions.Subject),
+                        new string[] { nameof(MonitorApiKeyOptions.Subject) }));
+            }
+
+            string jwkJson = null;
+            try
+            {
+                jwkJson = Base64UrlEncoder.Decode(sourceOptions.PublicKey);
+            }
+            catch (Exception)
+            {
+                errors.Add(
+                    new ValidationResult(
+                        string.Format(
+                            Strings.ErrorMessage_NotBase64,
+                            nameof(MonitorApiKeyOptions.PublicKey),
+                            sourceOptions.PublicKey),
+                        new string[] { nameof(MonitorApiKeyOptions.PublicKey) }));
+            }
+
+            JsonWebKey jwk = null; 
+            if (!string.IsNullOrEmpty(jwkJson))
+            {
+                try
+                {
+                    jwk = JsonWebKey.Create(jwkJson);
+                }
+                // JsonWebKey will throw only throw ArgumentException or a derived class.
+                catch (ArgumentException)
```

Comment:
Did you not want to propagate any of the exception message text?

Reply:
I suppose it's safe to propagate the message? I'm used to never doing that because it can expose unnecessary info.